File: src/main/java/io/github/gravetii/model/GameStats.java
package io.github.gravetii.model;

import io.github.gravetii.game.GameResult;
import io.github.gravetii.game.UserResult;

public class GameStats {
  private final GameResult gameResult;
  private final UserResult userResult;

  public GameStats(GameResult gameResult, UserResult userResult) {
    this.gameResult = gameResult;
    this.userResult = userResult;
  }

  @Override
  public String toString() {
    String result =
        "Words - %d/%d\n"
            + "Words percentage - %.2f\n\n"
            + "Score - %d/%d\n"
            + "Score percentage - %.2f";

    int userWordsCount = this.userResult.getWordCount();
    int totalWordsCount = this.gameResult.all().size();
    int userScore = this.userResult.getTotalScore();
    int totalScore = this.gameResult.getTotalScore();

    float wordsPercent = ((float) userWordsCount * 100) / totalWordsCount;
    float scorePercent = ((float) userScore * 100) / totalScore;

    return String.format(
        result,
        this.userResult.getWordCount(),
        gameResult.all().size(),
        wordsPercent,
        this.userResult.getTotalScore(),
        this.gameResult.getTotalScore(),
        scorePercent);
  }
}

File: src/main/kotlin/io/github/gravetii/model/GameStats.kt
package io.github.gravetii.model

import io.github.gravetii.game.GameResult
import io.github.gravetii.game.UserResult

data class GameStats(val gameResult: GameResult, val userResult: UserResult) {

    override fun toString(): String {
        val result = "Words - %d out of %d\n" +
                "Words percentage - %.2f\n\n" +
                "Score - %d out of %d\n" +
                "Score percentage - %.2f"

        val userWordsCount = userResult.getWordCount()
        val totalWordsCount = gameResult.all().size
        val userScore = userResult.totalScore
        val totalScore = gameResult.totalScore
        val wordsPercent = (userWordsCount * 100.0) / totalWordsCount
        val scorePercent = (userScore * 100.0) / totalScore
        return String.format(
            result,
            userResult.getWordCount(),
            gameResult.all().size,
            wordsPercent,
            userScore,
            totalScore,
            scorePercent
        )
    }
}